pkg/bgpv1/agent/controller.go

```diff
@@ -112,15 +137,23 @@ func NewController(params ControllerParams) (*Controller, error) {
 
 // Start is called by hive after all of our dependencies have been started.
 func (c *Controller) Start(startCtx hive.HookContext) error {
-	store, err := c.PolicyResource.Store(startCtx)
+	policyStore, err := c.PolicyResource.Store(startCtx)
 	if err != nil {
 		return fmt.Errorf("PolicyResource.Store(): %w", err)
 	}
 	c.PolicyLister = policyListerFunc(func() ([]*v2alpha1api.CiliumBGPPeeringPolicy, error) {
-		return store.List(), nil
+		return policyStore.List(), nil
+	})
+
+	ipStore, err := c.IPPoolResource.Store(startCtx)
+	if err != nil {
+		return fmt.Errorf("IPPoolResource.Store(): %w", err)
+	}
+	c.IPPooolLister = ipPoolListerFunc(func() ([]*v2alpha1api.CiliumPodIPPool, error) {
```

Comment:
`IPPooolLister` doesn't seem right?

Reply:
Thanks for the review @youngnick. Commit b18c075 implements this change request.